### PR TITLE
Adds note about 'mvn clean' in README.

### PR DESCRIPTION
**Pull request by coollog**
This lets users know that if they just run `mvn clean compile jib:build` each time, they don't get the cache speed-up.

### CODE REVIEW

**Comment by coollog** [comment]
@parkjam4 How does this sound?

File: README.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ Build your container image with:
 mvn compile jib:build
 ```
 
-Subsequent builds are much faster than the initial build.
+Subsequent builds are much faster than the initial build. `mvn clean` cleans Jib's build cache and forces Jib to re-pull the base image and re-build the application layers.
```
Comment (parkjam4) [inline README.md]: This new sentence seems disconnected from the rest of the section. 
Are you trying to say that `mvn compile jib:build` should be the command you run to build your image and not `mvn clean compile jib:build`?

Comment (coollog) [reply README.md]: Yes, the idea is that `mvn compile jib:build` is the command to run in most cases, and if you want to clear Jib's cache, `mvn clean` would do that.

Comment (parkjam4) [reply README.md]: Ok, what if you worded it like this:

Build your container image with:
 
 ```commandline
 mvn compile jib:build
 ```
Subsequent builds are much faster than the initial build. 

If you want to clear Jib's build cache and force it to re-pull the base image and re-build the application layers, run:

 ```commandline
 mvn clean compile jib:build
 ```

Comment (coollog) [reply README.md]: That sounds much clearer! Will update with this

**Pull request merged**